File: 백준/Silver/4375. 1/1.java
import java.io.*;

class Main {
    static int n, target, targetSize;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        while (true) {
            n = Integer.parseInt(br.readLine());
            target = 1;
            targetSize = 1;

            while (target % n != 0) {
                target = (target % n) * 10 + 1;
                targetSize++;
            }

            bw.append(String.valueOf(targetSize));
            bw.newLine();

            if (!br.ready()) break;
        }

        bw.flush();
        bw.close();
        br.close();
    }
}

File: 백준/Silver/4375. 1/1.kt
var targetSize = 0
var n = 0

fun main() {
    val br = System.`in`.bufferedReader()
    val sb = StringBuilder()

    while (true) {
        n = br.readLine().toInt()
        targetSize = 1

        while (!modTarget()) {
            targetSize++
        }

        sb.append(targetSize).append('\n')

        if (!br.ready()) break
    }

    println(sb)
    br.close()
}

fun modTarget(): Boolean {
    var r = 0

    repeat(targetSize) {
        r = ((r * 10) + 1) % n
    }

    return r == 0
}